### PR TITLE
feat(dataset): support dataset string id

### PR DESCRIPTION
**Pull request by tianweidut**
## Description
- server pr: https://github.com/star-whale/starwhale/pull/1429
- issue: https://github.com/star-whale/starwhale/issues/1419
- id is string: speech commands example
image

## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [x] Client
- [x] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/api/_impl/dataset/loader.py [hunk 1]
```diff
@@ -15,31 +13,37 @@
...
         self.start = start
         self.end = end
-        self.logger = logger or _logger
+
         # TODO: refactor TabularDataset with dataset_uri
         # TODO: refactor dataset, tabular_dataset and standalone dataset module
         self.tabular_dataset = TabularDataset.from_uri(
             dataset_uri, start=start, end=end
         )
+        self.session_consumption = session_consumption
```
Comment (jialeicui) [inline client/starwhale/api/_impl/dataset/loader.py:43]: Add range checking for loader and `session_consumption`?
Different ranges will cause unexpected behavior.

Comment (tianweidut) [reply client/starwhale/api/_impl/dataset/loader.py:43]: ok

Comment (tianweidut) [reply client/starwhale/api/_impl/dataset/loader.py:44]: done

**Review by goldenxinxing — APPROVED** [review]
LGTM

**Pull request merged**